src/main/kotlin/dsl/PersonBuilder.kt

```diff
@@ -0,0 +1,32 @@
+package dsl
+
+class PersonBuilder {
+    private lateinit var name: String
+    private var company: String? = null
+    private var skills: Skills? = null
+    private var languages: Languages? = null
+
+    fun name(value: String) {
+        name = value
+    }
+
+    fun company(value: String) {
+        company = value
+    }
+
+    fun skills(block: SkillsBuilder.() -> Unit) {
+        skills = SkillsBuilder().apply(block).build()
+    }
```

Comment:
지금은 `Skills`를 생성하는게 `fun skills(block: SkillsBuilder.() -> Unit)`만 있지만 다른 함수를 제공하게 되는 경우에는 `SkillsBuilder`를 메서드내에서 생성하는게 아니라 미리 생성된 프로퍼티 `SkillsBuilder`를 이용하도록 변경하시는게 `nullable`한 객체를 다루지 않아 안전할수도 있습니다~!
참고만 해주시면 좋을것 같아요!

```suggestion
class PersonBuilder {
    private lateinit var name: String
    private var company: String? = null
    private var skillsBuilder: SkillsBuilder = SkillsBuilder()
    private var languages: Languages? = null

    fun name(value: String) {
        name = value
    }

    fun company(value: String) {
        company = value
    }

    fun skills(block: SkillsBuilder.() -> Unit) {
        skillsBuilder.apply(block).build()
    }```